src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2588,6 +2574,30 @@ protected virtual MethodDefinition MarkMethod (MethodReference reference, Depend
 			return method;
 		}
 
+		void MarkForwardersInCopyAssembly (MemberReference memberReference, IMemberDefinition resolvedMember, DependencyInfo reason)
+		{
+			if (Annotations.GetAction (memberReference.Module.Assembly) != AssemblyAction.Copy)
+				return;
+
+			// If member comes from a type which was resolved from a forwarder in a copy assembly,
+			// mark the transitive chain of forwarders.
+			TypeReference typeRef = memberReference is TypeReference ? memberReference as TypeReference : memberReference.DeclaringType;
+			TypeDefinition resolvedTypeDef = resolvedMember is TypeDefinition ? resolvedMember as TypeDefinition : resolvedMember.DeclaringType;
+			foreach (var assembly in memberReference.Module.AssemblyReferences) {
+				if (assembly.MetadataToken == typeRef.Scope.MetadataToken) {
+					ModuleDefinition refModule = typeRef.Module.AssemblyResolver.Resolve (assembly).MainModule;
+					if (refModule.GetMatchingExportedType (resolvedTypeDef, out var exportedType)) {
+						MarkingHelpers.MarkExportedType (exportedType, refModule, reason);
+
+						if (resolvedTypeDef.IsNested)
```

Comment:
Regarding nested types: I believe the only typeref we need to check for forwarders is the outermost DeclaringType, since this is the one that could point to a type forward in another assembly. Instead of the recursive call for the declaring type (which will do unnecessary walks over the exported types), I would rather first get the outermost DeclaringType typeref, then resolve it and call MarkExportedType.